### PR TITLE
Add new structure updated

### PR DESCRIPTION
**Pull request by awirb**
this is an updated version of PR #433 - see also comments there
* now rebased on the master (all the secondary release changes)
*  also added a check for the variation dict 
*  and reading of flags section

### CODE REVIEW

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx [hunk 1]
```diff
@@ -127,7 +127,7 @@ def pointsToRasterC(x, y, z, Z0, csz=1, xllc=0, yllc=0):
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-def computeForceC(cfg, particles, fields, dem, dT):
+def computeForceC(cfg, particles, fields, dem, dT, frictType):
```
Comment (matthiasto) [inline avaframe/com1DFAPy/DFAfunctionsCython.pyx:130]:
``` suggestion
def computeForceC(cfg, particles, fields, dem, dT, int frictType):
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 2]
```diff
@@ -70,92 +74,49 @@ def com1DFAMain(cfg, avaDir, relThField):
...
-            entrainmentArea = inputSimLines['entrainmentArea']
-            resistanceArea = inputSimLines['resistanceArea']
-            reportDict = createReportDict(avaDir, logName, relName, inputSimLines['releaseLine'], cfgGen, entrainmentArea, resistanceArea, reportAreaInfo)
-            # add time and mass info to report
-            reportDict = reportAddTimeMassInfo(reportDict, tcpuDFA, cfgGen, infoDict)
+    if 'particles' in cfgGen['resType']:
+        # export particles dictionaries of saving time steps
+        outDirData = os.path.join(outDir, 'particles')
+        fU.makeADir(outDirData)
+        savePartToPickle(particlesList, outDirData)
```
Comment (matthiasto) [inline avaframe/com1DFAPy/com1DFA.py:104]: now logname is missing as input...

**Review by matthiasto — CHANGES_REQUESTED** [review]

File: avaframe/com1DFAPy/DFAfunctionsCython.pyx (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFAPy/DFAfunctionsCython.pyx]: done

File: avaframe/com1DFAPy/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py]: updated

File: avaframe/in1Data/getInput.py [hunk 3]
```diff
@@ -172,17 +170,20 @@ def getInputDataCom1DFAPy(avaDir, cfg, flagDev=False):
...
 
+    # Load dev flag as string
+    flagDev = cfg['flagDev']
+
     # Set flag if there is an entrainment or resistance area
     entResInfo= {'flagEnt': 'No', 'flagRes': 'No'}
 
     # Initialise release areas, default is to look for shapefiles
-    if flagDev is True:
+    if flagDev == 'True':
```
Comment (fso42) [inline avaframe/in1Data/getInput.py:180]: Similar to line 183, just use cfg['flagDev'] and drop line 173 and 174

**Review by fso42 — CHANGES_REQUESTED** [review]
As a note: as discussed, please remove the unnecessary for loop in deriveParamterset and renamen the overallConfig... to originConfig... or sourceConfig in the next PR

File: avaframe/in1Data/getInput.py (hunk 3 above)
Comment (awirb) [reply avaframe/in1Data/getInput.py:180]: Done

**Review by fso42 — APPROVED** [review]

**Review by matthiasto — APPROVED** [review]

**Pull request merged**